zerver/lib/rest.py

```diff
@@ -147,7 +147,7 @@ def rest_dispatch(request: HttpRequest, **kwargs: Any) -> HttpResponse:
             target_function = authenticated_rest_api_view(
                 allow_webhook_access="allow_incoming_webhooks" in view_flags,
             )(target_function)
-        elif request.path.startswith("/json") and "allow_anonymous_user_web" in view_flags:
+        elif request.path.startswith(("/json", "/avatar")) and "allow_anonymous_user_web" in view_flags:
```

Comment:
Used tuple here since I found out that we can use that here.

Reply:
Cool, that works.